activities.py

```diff
@@ -0,0 +1,72 @@
+# @@@SNIPSTART python-money-transfer-project-template-withdraw
+from temporalio import activity
+
+from banking_service import BankingService, InvalidAccountError
+from shared import PaymentDetails
+
+
+# Generic errors for activity failures
+class WithdrawFailedError(Exception):
+    pass
+
+
+class DepositFailedError(Exception):
+    pass
+
+
+class RefundFailedError(Exception):
+    pass
+
+
+@activity.defn
+async def withdraw(data: PaymentDetails) -> str:
+    reference_id: str = f"{data.reference_id}-withdrawal"
+    bank = BankingService("bank-api.example.com")
+    try:
+        confirmation = bank.withdraw(data.source_account, data.amount, reference_id)
+        return confirmation
+    except InvalidAccountError as error:
+        raise error
+    except Exception as error:
+        activity.logger.error(f"Withdrawal failed: {error}")
+        raise WithdrawFailedError("Failed to withdraw funds") from error
```

Comment:
What is the use case for wrapping these exceptions in a new type? Not that it's wrong, but it seems like a premature abstraction. We tell the user which activity failed when giving to workflow (and then to client when giving to client).

Reply:
Directly raised the caught exceptions without wrapping them in custom types.